Task1/Source.cpp

```diff
@@ -0,0 +1,237 @@
+#define _CRT_SECURE_NO_WARNINGS
+#include<stdlib.h>
+#include<iostream>
+#include<string.h>
+
+using namespace std;
+
+/*
+0-data
+1-pointer to the next elment
+*/
+
+void StringListInit(char ***list)
+{
+	*list = (char **)malloc(2 * sizeof(char*));
+	(*list)[0] = NULL;
+	(*list)[1] = NULL;
+}
+
+
+void StringListAdd(char** list, char* str){
+	int str_size = strlen(str) + 1;
+	if (!list[0]) {
+		cout << "list empty" << endl;
+		list[0] = (char*)malloc(str_size*sizeof(char));
+		strcpy(list[0], str);
+		cout << "added item with data:" << str << endl;
+		return;
+	}
+
+	char **tmp = (char**)malloc(2 * sizeof(char*));
+	tmp[0] = (char*)malloc(str_size * sizeof(char));
+	strcpy(tmp[0], str);
+	
+	tmp[1] = 0;
+	char **head = (list);
+	while (head[1]) {
```

Comment:
using while loop is not efficient here. It is better to "realloc".
For example: list = (char**)realloc(list, (ListSize + 2) * sizeof(char *));